Description: Unescaped left brace in regex leads to warnings with perl 5.26
Origin: https://svn.apache.org/viewvc/spamassassin/trunk/lib/Mail/SpamAssassin/PerMsgStatus.pm?r1=1708863&r2=1791010&diff_format=h
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=869408

Index: spamassassin-3.4.1/lib/Mail/SpamAssassin/PerMsgStatus.pm
===================================================================
--- spamassassin-3.4.1.orig/lib/Mail/SpamAssassin/PerMsgStatus.pm
+++ spamassassin-3.4.1/lib/Mail/SpamAssassin/PerMsgStatus.pm
@@ -916,16 +916,16 @@ sub get_content_preview {
     $str .= shift @{$ary};
   }
   undef $ary;
-  chomp ($str); $str .= " [...]\n";
 
   # in case the last line was huge, trim it back to around 200 chars
   local $1;
-  $str =~ s/^(.{,200}).*$/$1/gs;
+  $str =~ s/^(.{200}).+$/$1 [...]/gm;
+  chomp ($str); $str .= "\n";
 
   # now, some tidy-ups that make things look a bit prettier
-  $str =~ s/-----Original Message-----.*$//gs;
+  $str =~ s/-----Original Message-----.*$//gm;
   $str =~ s/This is a multi-part message in MIME format\.//gs;
-  $str =~ s/[-_\*\.]{10,}//gs;
+  $str =~ s/[-_*.]{10,}//gs;
   $str =~ s/\s+/ /gs;
 
   # add "Content preview:" ourselves, so that the text aligns
